server/src/main/java/org/apache/iotdb/db/engine/snapshot/SnapshotLogAnalyzer.java

```diff
@@ -55,25 +55,47 @@ public boolean hasNext() {
     }
   }
 
+  public String getSnapshotId() {
+    return snapshotId;
+  }
+
   /**
-   * @return The next pair of files recorded in the log. The left one is the path of source file,
-   *     the right one is the path of target file
+   * Return the total num of file in this snapshot.
+   *
+   * @return
    */
-  public Pair<String, String> getNextPairs() {
-    if (reader == null) {
-      return null;
+  public int getTotalFileCountInSnapshot() throws IOException {
+    reader.reset();
+    String currLine;
+    int cnt = 0;
+    while ((currLine = reader.readLine()) != null && !currLine.equals(SnapshotLogger.END_FLAG)) {
+      cnt++;
     }
-    try {
-      String fileInfo = reader.readLine();
-      String[] filesPath = fileInfo.split(SnapshotLogger.SPLIT_CHAR);
-      if (filesPath.length != 2) {
-        LOGGER.warn("Illegal file info: {} in snapshot log", fileInfo);
-        return null;
-      }
-      return new Pair<>(filesPath[0], filesPath[1]);
-    } catch (IOException e) {
-      LOGGER.error("Exception occurs when analyzing snapshot log", e);
-      return null;
+    return cnt;
+  }
+
+  /**
+   * Read the tail of the log file to see if the snapshot is complete.
+   *
+   * @return
+   */
+  public boolean isSnapshotComplete() throws IOException {
+    char[] endFlagInChar = new char[SnapshotLogger.END_FLAG.length()];
```

Comment:
Do the validation logic when constructing Analyzer to avoid multiple invoking to this method by others

Reply:
And you can use some symbol inside the analyzer to mark the result of validation

Reply:
Done